src/types/common.rs

```diff
@@ -258,6 +409,60 @@ impl Display for KeyMaterial {
     }
 }
 
+impl KeyMaterial {
+    pub const TAG: Tag = Tag::new(0x420043);
+
+    pub fn fast_scan(scanner: &mut FastScanner<'_>, format: &KeyFormatType) -> Result<Self, FastScanError> {
+        match format {
+            KeyFormatType::Raw
+            | KeyFormatType::Opaque
+            | KeyFormatType::PKCS1
+            | KeyFormatType::PKCS8
+            | KeyFormatType::X509 => scanner.scan_bytes(Self::TAG).map(|s| Self::Bytes(s.into())),
+
+            KeyFormatType::TransparentSymmetricKey => {
+                TransparentSymmetricKey::fast_scan(scanner).map(Self::TransparentSymmetricKey)
+            }
+            KeyFormatType::TransparentDSAPrivateKey => {
+                TransparentDSAPrivateKey::fast_scan(scanner).map(Self::TransparentDSAPrivateKey)
+            }
+            KeyFormatType::TransparentDSAPublicKey => {
+                TransparentDSAPublicKey::fast_scan(scanner).map(Self::TransparentDSAPublicKey)
+            }
+            KeyFormatType::TransparentRSAPrivateKey => {
+                TransparentRSAPrivateKey::fast_scan(scanner).map(Self::TransparentRSAPrivateKey)
+            }
+            KeyFormatType::TransparentRSAPublicKey => {
+                TransparentRSAPublicKey::fast_scan(scanner).map(Self::TransparentRSAPublicKey)
+            }
+            KeyFormatType::TransparentDHPrivateKey => {
+                TransparentDHPrivateKey::fast_scan(scanner).map(Self::TransparentDHPrivateKey)
+            }
+            KeyFormatType::TransparentDHPublicKey => {
+                TransparentDHPublicKey::fast_scan(scanner).map(Self::TransparentDHPublicKey)
+            }
+
+            _ => todo!(),
+        }
+    }
+
+    pub fn format(&self, formatter: &mut Formatter<'_>) -> FormatResult {
+        match self {
+            KeyMaterial::Bytes(v) => formatter.format_bytes(Self::TAG, v),
```

Comment:
I think this is wrong as it will write tag 0x420043 when it should write 0x420042.